src/resolvers/AstExplorerResolver.ts

```diff
@@ -0,0 +1,58 @@
+import { readFile, writeFile } from 'mz/fs';
+import { URL } from 'url';
+import NetworkResolver from './NetworkResolver';
+
+const EDITOR_URL_PATTERN = /^https:\/\/astexplorer\.net\/#\/gist\/(\w+)(?:\/(\w+))?$/;
+
+/**
+ * Resolves plugins from AST Explorer transforms.
+ *
+ * astexplorer.net uses GitHub gists to save and facilitate sharing. This
+ * resolver accepts either the editor URL or the gist API URL.
+ */
+export default class AstExplorerResolver extends NetworkResolver {
+  constructor(
+    private readonly baseURL: URL = new URL('https://astexplorer.net/')
```

Comment:
This parameter is only used in tests so we can test against a local server instead of the real astexplorer.net.